pkg/api/internalclient/device.go

```diff
@@ -82,7 +82,7 @@ func (c *client) DeviceLookup(lookup map[string]string) (*models.Device, []error
 	}
 
 	if resp.StatusCode() != http.StatusOK {
-		return nil, []error{err}
+		return nil, []error{errors.New("failed to get the device on the API")}
```

Comment:
Shouldn't `err` be included in the error?

Reply:
In this case, the error in this arm is nil, that why I have created this hard-coded error.

Reply:
```suggestion
		return nil, []error{errors.New("fail to get the device from the API")}
```